mail/postfix/Makefile

```diff
@@ -177,136 +181,78 @@ newaliases_path=/usr/bin/newaliases
 mailq_path=/usr/bin/mailq
 shlib_directory=/usr/lib/postfix
 meta_directory=/etc/postfix
+# postinit script may override the values below if the external
+# drive is at a mountpoint other than /data/
+data_directory=/data/postfix/lib
+queue_directory=/data/postfix/spool
+mail_spool_directory=/data/postfix/mail
 
 ln_suffix=.postfix
-ln_old_suffix=.old
+ln_suffix_pre_pf=.pre-postfix
```

Comment:
I think it is worth it because ".old" is too generic. It is very difficult to identify what package or script renamed a file to ".old". One must guess that the renaming was part of the installation process of some package, and then sift through the files in /usr/lib/opkg/info/ to identify the package that renamed the file. I think in many cases a person might even think that it was he/she who might have renamed the file manually while experimenting with something in the file system.

IMO the approach of saving the pre-existing versions of sendmail, mailq and newaliases executables is laudable. Other packages (msmtp, for one) install executables without checking, resulting in build errors where another package attempts to install the same file. 

If postrm fails to restore ".old" to the original name, the user will clearly see the problem because mailq, sendmail or newaliases won't execute properly. The user can then fix the problem by re-installing the appropriate package (like msmtp).

All of this should become obsolete with the [introduction of alternatives](http://lists.infradead.org/pipermail/lede-dev/2017-March/006496.html) in LEDE/OpenWrt. At some later point, I am going to propose patches to msmtp, postfix and sendmail (which is still PR #4369) to enable the alternatives support if it's not already done by package maintainers by then.

Reply:
Ok, I agree with your point that sendmail.pre-postfix is more clear than sendmail.old, but could we at least add a rename in a postinst script from sendmail.old to sendmail.pre-postfix and so on?

Reply:
I agree, postrm can also check for presence of ".old" files and rename them if they are present.

I'll wait for your other comments and make all revisions in one commit.

Reply:
Actually, yes, postinst can also rename ".old" into ".pre-postfix".